src/cli/java/org/commcare/util/screen/QueryScreen.java

```diff
@@ -191,12 +196,23 @@ public Pair<ExternalDataInstance, String> processResponse(InputStream responseDa
         return instanceOrError;
     }
 
-    public void setQueryDatum(ExternalDataInstance dataInstance) {
+    public void updateSession(ExternalDataInstance dataInstance) {
         if (dataInstance != null) {
-            sessionWrapper.setQueryDatum(dataInstance);
+            ExternalDataInstance userInputInstance = getUserInputInstance();
+            sessionWrapper.setQueryDatum(dataInstance, userInputInstance);
```

Comment:
I realize that this is going to create a new instance on every request until the form session starts. Will need to think about how to prevent that

Reply:
Also see this discussion about instance IDs & refs: https://github.com/dimagi/commcare-core/pull/1082#issuecomment-1134448655

Reply:
[f2d6616](https://github.com/dimagi/commcare-core/pull/1112/commits/f2d6616c292b09ff2ad86bd7804e60f710cfdccd)